pyscf/mcscf/ucasci.py

```diff
@@ -296,13 +296,13 @@ def _finalize(self):
             ss_core = self._scf.spin_square(mocore, ovlp_ao)
             if isinstance(self.e_cas, (float, numpy.number)):
                 ss = fci.spin_op.spin_square(self.ci, self.ncas, self.nelecas,
-                                             mocas, ovlp_ao)
+                                             mocas, ovlp_ao, self.fcisolver)
```

Comment:
How about calling `self.fcisolver.spin_square()` here?

Reply:
I think calling ``self.fcisolver.spin_square()`` will break the original code even when the ``self.fcisolver`` is a normal ``direct_spin1`` FCI solver. The reasons are the following:

(1) Near the calling position of ``fci.spin_op.spin_square`` there is an ``if`` statement: https://github.com/pyscf/pyscf/blob/master/pyscf/mcscf/ucasci.py#L288 and the ``if`` checks that there is indeed a ``spin_square`` method. But then the original code **deliberately** does not call that ``spin_square`` method.

(2) We can also check the implementation of ``self.fcisolver.spin_square()`` in ``direct_spin1`` FCI solver: https://github.com/pyscf/pyscf/blob/master/pyscf/fci/direct_spin1.py#L829-L832 and it calls ``fci.spin_op.spin_square0`` and the documentation for it says that ``Spin square for RHF-FCI CI wfn only``. So it may indicate that calling ``self.fcisolver.spin_square()`` in UCASCI will not give the correct behavior.